### PR TITLE
FIX: Bugs in algorithm, missing solver metadata

### PR DESCRIPTION
**Pull request by tasseff**
This addresses some issues I was encountering while testing the new version. Not sure if there's a more accurate way to set `PrimalStatus` and `ResultCount`.

### CODE REVIEW

File: src/algorithm.jl [hunk 1]
```diff
@@ -5,13 +5,15 @@
 #  file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 eval_func(values, func) = MOI.Utilities.eval_variables(vi -> values[vi.value], func)
+
 function eval_objective(model::Optimizer, values)
-    return if model.nlp_block.has_objective
-        MOI.eval_objective(model.nlp_block.evaluator, values)
+    if model.nlp_block != nothing && model.nlp_block.has_objective
```
Comment (blegat) [inline src/algorithm.jl:10]: Use `!==`

**Comment by blegat** [comment]
Do you still plan to change something ? You marked the PR as draft but it seems to be good to merge. The coverage for your lines will come with https://github.com/JuliaOpt/Pavito.jl/issues/25 or https://github.com/JuliaOpt/Pavito.jl/issues/22 but these can be done in separate PRs

**Comment by tasseff** [comment]
Feel free to merge. I'll continue experimenting once the Ipopt.jl `is_valid` function is implemented.

**Comment by blegat** [comment]
@tasseff should be working on Ipopt.jl v0.6.3

**Pull request merged**